packages/shared-components/src/components/onto-dropdown/onto-dropdown.tsx

```diff
@@ -0,0 +1,128 @@
+import {Component, Event, h, State, Prop, EventEmitter, Listen} from '@stencil/core';
+import {DropdownItem} from '../../models/dropdown/dropdown-item';
+import {TranslationService} from '../../services/translation.service';
+import {DropdownItemAlignment} from '../../models/dropdown/dropdown-item-alignment';
+
+/**
+ * A reusable dropdown component built using StencilJS. This component supports configurable labels, tooltips, icons,
+ * and items, making it versatile for various use cases. It also integrates with a translation service to handle
+ * internationalization.
+ */
+@Component({
+    tag: 'onto-dropdown',
+    styleUrl: 'onto-dropdown.scss',
+    shadow: false,
+})
+export class OntoDropdown {
+
+    /**
+     * Indicates whether the dropdown menu is open.
+     */
+    @State() open = false;
+
+    /**
+     * The dropdown button name. It will be used if present; otherwise, the {@link OntoDropdown#dropdownButtonNameLabelKey} will be used.
+     */
+    @Prop() dropdownButtonName: string;
+
+    /**
+     * The translation label key for the dropdown button name. It will be used if {@link OntoDropdown#dropdownButtonName} is not present.
+     */
+    @Prop() dropdownButtonNameLabelKey: string;
+
+    /**
+     * The dropdown button tooltip. It will be used if present; otherwise, the {@link OntoDropdown#dropdownButtonTooltipLabelKey} will be used.
+     */
+    @Prop() dropdownButtonTooltip: string;
+
+    /**
+     * The translation label key for the dropdown button tooltip. It will be used if {@link OntoDropdown#dropdownButtonTooltip} is not present.
+     */
+    @Prop() dropdownButtonTooltipLabelKey: string;
+
+    /**
+     * Icon class for the main dropdown button.
+     */
+    @Prop() iconClass = '';
+
+    /**
+     * Array of dropdown options.
+     */
+    @Prop() items: DropdownItem[];
+
+    /**
+     * Specifies the dropdown items' alignment. If not provided, the items and the dropdown button will be aligned to the left.
+     *
+     */
+    @Prop() dropdownAlignment: DropdownItemAlignment = DropdownItemAlignment.LEFT;
```

Comment:
I think it'd be better if we use plain string options here instead of enum values because we don't export the module types out of this module. Or, consider exporting the models as well, although I'm not sure if really want to do this because some of these are for internal use only.